### ISSUE TITLE
4-variant comparison 実測と Gate 2 Go/No-Go 判定

### ISSUE THREAD

**Issue by Kewton** [Kewton/photon-mlx#20]
## 概要

Issue #5 で実装した 4-variant comparison framework (`bench/run_all.py`) を使って、実際の 4 variant 計測を実行し、Gate 2 の Go/No-Go 判定を行う。

Issue #10 Epic のロードマップにおける**最終ステップ**。これを経て Gate 2 判定が完了する。

## 前提条件

- ✅ Issue #7 (retrieval 修正) 完了 — PR #11
- ✅ Issue #1 (no-citation Phase 1) 完了 — PR #13
- ✅ Issue #2 (full eval) 完了 — baseline 実測済み
- ✅ Issue #3 (E2E 統合) 完了 — PR #18 (PhotonRAGPipeline)
- ✅ Issue #4 (PHOTON small 学習) 完了 — PR #17 (gradient accumulation)
- ✅ Issue #5 (Gate 2 framework) 完了 — PR #19 (`bench/run_all.py`)
- ✅ Issue #8 (tokenizer 移行) / #9 (生成品質検証) 完了

## 計測対象の 4 variant

| Variant | 構成 | 目的 |
|---------|------|------|
| A. baseline | Baseline RAG のみ | 下限ライン |
| B. baseline + memory | Baseline RAG + summary memory 強化 | 中間コントロール |
| C. photon-rag | PHOTON session inference + baseline retrieval | 提案手法 |
| D. photon-rag + safe-recgen | C + Safe RecGen fallback | 完全版 |

### 前処理（必要なら）
1. PHOTON small (~200M) の学習チェックポイントを用意
   - 未学習なら `scripts/train_photon.py --config configs/photon_small.yaml` で学習
2. 既存インデックスが使えることを確認（chunks.db, lexical.pkl, embedding/, symbol_graph.json）

### Gate 2 判定ラン
```bash
python bench/run_all.py \
  --config configs/eval.yaml \
  --variants baseline,baseline_memory,photon_rag,photon_rag_safe_recgen \
  --static-eval data/eval_sets/static_eval.jsonl \
  --multi-turn-eval data/eval_sets/multi_turn_eval.jsonl
```

### 集計・レポート
```bash
# 各 variant の run_id を使って集計
python scripts/export_report.py --run-id <variant_run_id>
```

### Gate 2 判定基準（spec §19）

| 条件 | 必要値 | 判定方法 |
|------|--------|----------|
| PHOTON forward が安定 | stable | test_forward.py 全パス |
| tiny/small で学習が回る | loss が下がる | 既に確認済み (tiny で loss 10.56→3.13) |
| drift 指標が取得できる | 3 metrics | latent_cosine_drift / token_agreement / logit_kl |
| **follow-up 改善の兆候** | baseline 比 -30% latency | **本 Issue で測定** |

### 計測値ターゲット

| 指標 | Variant A (baseline) | Variant D (photon+recgen) | 目標 |
|------|---------------------|----------------------------|------|
| MT follow-up latency P50 | 実測 | 実測 | D < A × 0.7 |
| MT no-citation rate | 25% | 実測 | D ≤ A |
| Wrong citation rate | 0% | 実測 | D ≤ A |
| Memory P50 | 19 MB | 実測 | D ≤ A × 2 |

### 最終判定

- **Go**: 全 Gate 2 条件を満たし、かつ Variant D の latency 改善が baseline 比 -30% 以上
- **No-Go**: いずれかの条件を満たさない → scope 縮小判断 (spec §20)
  - PHOTON の scope を縮小し、summary memory 強化を control として継続
  - baseline 単体をプロダクトラインとして維持

## 既知のリスク

1. **PHOTON small の学習**: 未完了の場合は事前に学習が必要（~1日）
2. **4 variant × 180 turns の実行時間**: 推定 4-8 時間
3. **Variant D (Safe RecGen) の fallback 頻度**: Safe RecGen 発火率が高すぎると latency が悪化
4. **Wrong citation の増加リスク**: PHOTON session inference で memory が stale になるケース

## 関連ファイル

- `bench/run_all.py` — 4-variant runner
- `configs/eval.yaml` — eval 設定
- `baseline_reporag/photon_pipeline.py` — PhotonRAGPipeline
- `photon_mlx/safe_recgen.py` — Safe RecGen controller
- `scripts/export_report.py` — レポート出力
- `reports/benchmark_report.md` — 最終レポート更新先
- `reports/failure_cases.md` — variant 別 failure cases 追加

## 関連 Issue

- Epic #10 (クローズ済み): Gate 2 到達ロードマップ
- #5 (クローズ済み): Comparison matrix framework
- #3 (クローズ済み): E2E 統合
- #14 (オープン): No-citation Phase 2（Gate 2 未達の場合の追加対策）

## 成果物

1. `logs/<variant_run_id>.jsonl` × 4（各 variant の run log）
2. `reports/benchmark_report.md` の Comparison Matrix 更新
3. `reports/gate2_judgment.md` — Go/No-Go 判定レポート（新規作成）
4. `reports/failure_cases.md` — variant 別 failure 分析追加

**Comment by Kewton** [comment]
4-variant 実測完了。Gate 2 判定: **No-Go**。詳細は reports/gate2_judgment.md を参照。

## 主要結果

### Static Eval 120問

| Variant | No-citation | P50 latency |
|---------|-------------|-------------|
| baseline_rag | 44.2% | 11,291 ms |
| baseline_rag + summary_memory | 40.0% | 12,168 ms |
| photon_rag | 42.5% | 11,738 ms |
| photon_rag + safe_recgen | 40.8% | 12,566 ms |

### Gate 2 判定: **No-Go**
- 必須条件「follow-up latency baseline 比 -30%」は PHOTON MT バグにより測定不可
- Static では 4 variants 間に実効差なし

### 推奨
- baseline_rag をプロダクトラインとして採用
- PHOTON scope を縮小
- 次: Issue #14 (No-citation Phase 2 post-processing) で追加改善
